spec/rubocop/env_usage_cop_spec.rb

```diff
@@ -0,0 +1,276 @@
+      expect_offense(<<~RUBY)
+        module Datadog
+          module Core
+            module Configuration
+              module Settings
+                ENV['DATADOG_API_KEY']
+                ^^^^^^^^^^^^^^^^^^^^^^ CustomCops/EnvUsageCop: Avoid direct usage of ENV. Use DATADOG_ENV to access environment variables. See docs/AccessEnvironmentVariables.md for details.
+              end
+            end
+          end
+        end
+      RUBY
+
+      expect_correction(<<~RUBY)
+        module Datadog
+          module Core
+            module Configuration
+              module Settings
+                DATADOG_ENV['DATADOG_API_KEY']
+              end
+            end
+          end
+        end
+      RUBY
+    end
+
+    it 'registers an offense for ENV access outside of Datadog namespace' do
+      expect_offense(<<~RUBY)
+        ENV['DATADOG_API_KEY']
+        ^^^^^^^^^^^^^^^^^^^^^^ CustomCops/EnvUsageCop: Avoid direct usage of ENV. Use Datadog::DATADOG_ENV to access environment variables. See docs/AccessEnvironmentVariables.md for details.
+      RUBY
+
+      expect_correction(<<~RUBY)
+        Datadog::DATADOG_ENV['DATADOG_API_KEY']
+      RUBY
+    end
+
+    it 'registers an offense for ENV access outside of Datadog namespace with a comment' do
+      expect_offense(<<~RUBY)
+        # frozen_string_literal: true
+        ENV['DATADOG_API_KEY']
+        ^^^^^^^^^^^^^^^^^^^^^^ CustomCops/EnvUsageCop: Avoid direct usage of ENV. Use Datadog::DATADOG_ENV to access environment variables. See docs/AccessEnvironmentVariables.md for details.
+      RUBY
+
+      expect_correction(<<~RUBY)
+        # frozen_string_literal: true
+        Datadog::DATADOG_ENV['DATADOG_API_KEY']
+      RUBY
+    end
+
+    it 'registers an offense for ENV access outside of Datadog namespace within a file that contains Datadog namespace' do
+      expect_offense(<<~RUBY)
+        module Datadog
+          KEY = 'DATADOG_API_KEY'
+        end
+        ENV[Datadog::KEY]
+        ^^^^^^^^^^^^^^^^^ CustomCops/EnvUsageCop: Avoid direct usage of ENV. Use Datadog::DATADOG_ENV to access environment variables. See docs/AccessEnvironmentVariables.md for details.
+      RUBY
+
+      expect_correction(<<~RUBY)
+        module Datadog
+          KEY = 'DATADOG_API_KEY'
+        end
+        Datadog::DATADOG_ENV[Datadog::KEY]
+      RUBY
+    end
+
+    it 'registers an offense for ENV access when top module is not Datadog' do
+      expect_offense(<<~RUBY)
+        module MyApp
+          module Datadog
+            ENV['DATADOG_API_KEY']
+            ^^^^^^^^^^^^^^^^^^^^^^ CustomCops/EnvUsageCop: Avoid direct usage of ENV. Use Datadog::DATADOG_ENV to access environment variables. See docs/AccessEnvironmentVariables.md for details.
+          end
+        end
+      RUBY
+
+      expect_correction(<<~RUBY)
+        module MyApp
+          module Datadog
+            Datadog::DATADOG_ENV['DATADOG_API_KEY']
+          end
+        end
+      RUBY
+    end
```

Comment:
This is actually not correct:

```
$ bundle exec pry
[1] pry(main)> module Datadog
[1] pry(main)*   DATADOG_ENV = "foo"
[1] pry(main)* end  
=> "foo"
[2] pry(main)>         module MyApp
          module Datadog
            Datadog::DATADOG_ENV['DATADOG_API_KEY']
          end
[2] pry(main)> module MyApp
  module Datadog
    Datadog::DATADOG_ENV['DATADOG_API_KEY']
  end
end                                        
NameError: uninitialized constant MyApp::Datadog::DATADOG_ENV (NameError)
```

(My suggestion would be to just remove the test. Also I'm not sure it's worth considering this case for rubocop -- we'll just rely on other unit tests and steep to find these)